radixdlt/src/main/java/com/radixdlt/middleware2/ClientAtom.java

```diff
@@ -90,15 +92,18 @@ private ClientAtom(
 	}
 
 	@JsonProperty("raw")
-	private void setSerializerAtom(byte[] atomBytes) {
-		Objects.requireNonNull(atomBytes);
+	private void setSerializerAtom(byte[] rawAtom) {
+		Objects.requireNonNull(rawAtom);
+		this.rawAtom = rawAtom;
+	}
+
+	public void init(Hasher hasher, Serialization serialization) {
```

Comment:
how about doing it this way then (introducing an intermediate class):
```
class RawClientAtom {
   byte[] rawAtom;

   public ClientAtom toClientAtom() {
     return new ClientAtom(...init from raw data here...);
  }
}
```

Reply:
I actually think a bit of this will be resolved in my PR: https://github.com/radixdlt/radixdlt-core/pull/231 as the hash itself will get serialized.

I think cleanest way to do this is to merge that branch into this one.